File: app/models/vegan_company.rb
# == Schema Information
#
# Table name: vegan_companies
#
#  id          :bigint           not null, primary key
#  name        :string
#  slug        :string
#  website     :string
#  description :text
#  created_at  :datetime         not null
#  updated_at  :datetime         not null
#
class VeganCompany < ApplicationRecord
end

File: src/VegDex.Core/Entities/VeganCompany.cs
namespace VegDex.Core.Entities;

public class VeganCompany : Entity
{
    public string? Description { get; set; }
    public string Name { get; set; }
    public string Slug { get; set; }
    public string? Website { get; set; }
    public static VeganCompany Create(
        int veganCompanyId,
        string name,
        string? description = null,
        string? website = null)
    {
        var veganCompany = new VeganCompany
        { Id = veganCompanyId,
          Name = name,
          Description = description,
          Website = website };
        return veganCompany;
    }
}
